src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1590,6 +1590,45 @@ protected override void HandleSetBreakpointsRequestAsync(IRequestResponder<SetBr
                     var resBreakpoints = new List<Breakpoint>();
                     foreach (var bp in breakpoints)
                     {
+                        if (dict.ContainsKey(bp.Line))
+                        {
+                            // already created
+                            IDebugBreakpointRequest2 breakpointRequest;
+                            if (dict[bp.Line].GetBreakpointRequest(out breakpointRequest) == 0)
+                            {
+                                var ad7BPRequest = (AD7BreakPointRequest)breakpointRequest;
```

Comment:
In case we end up supporting other types of breakpoints, it might be better to add in the if statement `&& breakpointRequest is AD7BreakPointRequest ad7BPRquest)`

Reply:
I realize you copied this from where it was below but i think it might be worth fixing now with the new language changes.